sklearn/metrics/tests/test_classification.py

```diff
@@ -955,6 +988,96 @@ def test_multilabel_jaccard_similarity_score():
     assert_equal(jaccard_similarity_score(y1, np.zeros(y1.shape)), 0)
     assert_equal(jaccard_similarity_score(y2, np.zeros(y1.shape)), 0)
 
+    y_true = np.array([[0, 1, 1], [1, 0, 0]])
+    y_pred = np.array([[1, 1, 1], [1, 0, 1]])
+    # average='macro'
+    assert_almost_equal(jaccard_similarity_score(y_true, y_pred,
+                                                average='macro'), 2. / 3)
+    # average='micro'
+    assert_almost_equal(jaccard_similarity_score(y_true, y_pred,
+                                                average='micro'), 3. / 5)
+    # average='samples' (default)
+    assert_almost_equal(jaccard_similarity_score(y_true, y_pred), 7. / 12)
+    assert_almost_equal(jaccard_similarity_score(y_true, y_pred,
+                                sample_weight=np.array([0.1, 0.9])), 31. / 60)
+    assert_almost_equal(jaccard_similarity_score(y_true, y_pred,
+                                                 average='samples',
+                                                 labels=[0, 2]), 1. / 2)
+    assert_almost_equal(jaccard_similarity_score(y_true, y_pred,
+                                                 average='samples',
+                                                 labels=[1, 2]), 1. / 2)
+    assert_almost_equal(jaccard_similarity_score(y_true, y_pred,
+                                                 average='samples',
+                                                 sample_weight=[1, 2]), 5. / 9)
+    assert_almost_equal(jaccard_similarity_score(y_true, y_pred,
+                                                 average='micro',
+                                                 sample_weight=[1, 2]), 4. / 7)
+    y_true = np.array([[0, 1, 1], [1, 0, 1]])
+    y_pred = np.array([[1, 1, 1], [1, 0, 1]])
+    assert_almost_equal(jaccard_similarity_score(y_true, y_pred,
+                                                 average='macro'), 5. / 6)
+    assert_almost_equal(jaccard_similarity_score(y_true, y_pred,
+                                                 average='macro',
+                                                 sample_weight=[1, 2]), 8. / 9)
+    assert_almost_equal(jaccard_similarity_score(y_true, y_pred,
+                                                 average='weighted'), 7. / 8)
+
+
+def test_multiclass_jaccard_similarity_score():
+    y_true = ['ant', 'ant', 'cat', 'cat', 'ant', 'cat', 'bird', 'bird']
```

Comment:
Just make this test check that multiclass results are equivalent to using LabelBinarizer on the multiclass data to make it multilabel. I.e. you never need to be comparing to exact expected results. Just create a helper that asserts the invariance that holds between multclass and multilabel.